pkg/kube/kube_handler.go

```diff
@@ -586,13 +589,63 @@ func doValidationRequest(cli HTTPClient, req *http.Request) (*ValidationResponse
 	return &ValidationResponse{Code: res.StatusCode, Message: string(body)}, nil
 }
 
+func getRequests(appRepo *v1alpha1.AppRepository, cli HTTPClient) ([]*http.Request, error) {
+	result := []*http.Request{}
+	repoURL := strings.TrimSuffix(strings.TrimSpace(appRepo.Spec.URL), "/")
+
+	switch appRepo.Spec.Type {
+	case "oci":
+		// For the OCI case, we want to validate that all the given repositories are valid
+		if len(appRepo.Spec.OCIRepositories) == 0 {
+			return nil, ErrEmptyOCIRegistry
+		}
+		for _, repoName := range appRepo.Spec.OCIRepositories {
+			parsedURL, err := url.ParseRequestURI(repoURL)
+			if err != nil {
+				return nil, err
+			}
+			parsedURL.Path = path.Join("v2", parsedURL.Path, repoName, "tags", "list")
+			q := parsedURL.Query()
+			q.Add("n", "1")
+			parsedURL.RawQuery = q.Encode()
+			req, err := http.NewRequest("GET", parsedURL.String(), nil)
```

Comment:
Does `oras` not provide helpers for getting the tag list of a repo? Ah, or is this so both OCI and non-oci repos can be validated by a bunch of plain requests?

Reply:
nope, `oras` only exposes methods to push and pull artifacts, not to interact with an OCI API (probably there is some package for that, but this is simple enough to just use the API with a request).